### DISCUSSION TITLE
Add .root modifier to x-intersect

### DISCUSSION THREAD

**Discussion by digitall-it** [alpinejs/alpine#2993] [3. Feature ideas]
If x-intersect is used on a sticky header table to detect when a cell comes into view, it may lead to unexpected result, for the fact that the IntersectionObserver seems to be always initialized to check for intersection related to the page element, and there doesn't seem to be a way to specify the root. I would like to implement it into a Livewire component so I had to resort to putting the observer initializer into x-data.

**Comment by SimoTod** [comment]
Do you have a specific syntax in mind? It might not be simple to define that behaviour with a modifier unless we accept that the root will always be an x-ref available from the current component.

**Reply by digitall-it** [reply]
I was thinking about traversing to closest, for example I used this workaround code and you can see that root is `this.$el.closest(selector)`. I think we may assume that anything more complex than that could be difficult to manage. Also note that I had to unobserve the intersection as I know that upon @this.call a new observer will be placed down the line and I had the old one still firing (but that's a different problem).

```
x-data="
{
    observe_next()
    {
        let observer_next = new IntersectionObserver((entries, observer) => {
            entries.forEach(entry => {
                if (entry.isIntersecting) {
                    observer.unobserve(entry.target);
                    console.log('intersecting next ' + this.$el.innerHTML);
                    @this.call('nextWeek')
                }
            })
        }, {
            root: this.$el.closest('.workshift_grid'),
        })

        observer_next.observe(this.$el)
    }
}
                                    "
                                x-init="observe_next"
```

So I was thinking

```<div x-intersect="func" x-data="x-intersect-root: selector">...</div>```

or

```<div x-intersect="func" x-intersect-data="root: selector">...</div>```

that's because it may be impossible to follow a shorthand syntax, like the rootMargin one, because a selector can contain spaces and dots.

With x-intersect-data you could pass all the modifiers at once as it is evaluated as a key value array and it's easy to read:

```<div x-intersect="func" x-intersect-data="root: 'table.workshift_grid', margin: '200px 50px 0px 0px', threshold: 'full' ">...</div>```
